MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift

```diff
@@ -160,39 +119,91 @@ private extension FarmerDetailsView {
                 }
                 Spacer()
                 phoneButton
-                    .frame(width: 80, height: 35)
-                    .background(.ultraThinMaterial)
-                    .clipShape(.capsule)
-                    .padding()
-                    .shadow(radius: 4)
             }
             Divider()
-            VStack(alignment: .leading) {
-                Text("Email")
-                    .foregroundStyle(.secondary)
-                    .font(.subheadline)
-                Link(viewModel.email!, destination: URL(string: viewModel.email!)!)
+            HStack {
+                VStack(alignment: .leading) {
+                    Text("Email")
+                        .foregroundStyle(.secondary)
+                        .font(.subheadline)
+                    Text(viewModel.email ?? "Aucune adresse email")
+                        .font(.callout)
+                }
+                Spacer()
+                mailButton
             }
             Divider()
-            VStack(alignment: .leading) {
+            VStack(alignment: .leading, spacing: 5) {
+                Text("Site Web")
+                    .foregroundStyle(.secondary)
+                    .font(.subheadline)
                 if !viewModel.websites.isEmpty {
-                    Text("Site Web")
-                        .foregroundStyle(.secondary)
-                        .font(.subheadline)
                     ForEach(viewModel.websites, id: \.id) { website in
-                        Link(website.websiteType.name, destination: URL(string: website.url)!)
+                        Text(.init("[\(website.websiteType.name)](\(website.url))"))
                     }
+                } else {
+                    Text("Aucun site web")
+                        .font(.callout)
                 }
             }
         }
         .padding()
         .background(.ultraThinMaterial)
         .clipShape(.rect(cornerRadius: 20))
+        .shadow(radius: 8)
+    }
+
+    private var directionButton: some View {
+        Button {
+            viewModel.onItineraryButtonTapped()
```

Comment:
onDirectionButtonTapped or itineraryButton ? 😅

Reply:
onDirectionButtonTapped of course